Description: fix program names after --program-transform-name
Author: Azer Abdullaev (Like All) <like-all@yandex.com>

--- systemd-221.orig/Makefile.am
+++ systemd-221/Makefile.am
@@ -182,18 +182,18 @@ AM_CPPFLAGS = \
 	-DUSER_DATA_UNIT_PATH=\"$(userunitdir)\" \
 	-DCERTIFICATE_ROOT=\"$(CERTIFICATEROOT)\" \
 	-DCATALOG_DATABASE=\"$(catalogstatedir)/database\" \
-	-DSYSTEMD_CGROUP_AGENT_PATH=\"$(rootlibexecdir)/systemd-cgroups-agent\" \
+	-DSYSTEMD_CGROUP_AGENT_PATH=\"$(rootlibexecdir)/cgroups-agent\" \
 	-DSYSTEMD_BINARY_PATH=\"$(rootlibexecdir)/systemd\" \
-	-DSYSTEMD_FSCK_PATH=\"$(rootlibexecdir)/systemd-fsck\" \
-	-DSYSTEMD_SHUTDOWN_BINARY_PATH=\"$(rootlibexecdir)/systemd-shutdown\" \
-	-DSYSTEMD_SLEEP_BINARY_PATH=\"$(rootlibexecdir)/systemd-sleep\" \
+	-DSYSTEMD_FSCK_PATH=\"$(rootlibexecdir)/fscheck\" \
+	-DSYSTEMD_SHUTDOWN_BINARY_PATH=\"$(rootlibexecdir)/shutdown\" \
+	-DSYSTEMD_SLEEP_BINARY_PATH=\"$(rootlibexecdir)/sleep\" \
 	-DSYSTEMCTL_BINARY_PATH=\"$(rootbindir)/systemctl\" \
-	-DSYSTEMD_TTY_ASK_PASSWORD_AGENT_BINARY_PATH=\"$(rootbindir)/systemd-tty-ask-password-agent\" \
-	-DSYSTEMD_STDIO_BRIDGE_BINARY_PATH=\"$(bindir)/systemd-stdio-bridge\" \
+	-DSYSTEMD_TTY_ASK_PASSWORD_AGENT_BINARY_PATH=\"$(rootbindir)/tty-ask-password-agent\" \
+	-DSYSTEMD_STDIO_BRIDGE_BINARY_PATH=\"$(bindir)/stdio-bridge\" \
 	-DROOTPREFIX=\"$(rootprefix)\" \
 	-DRANDOM_SEED_DIR=\"$(localstatedir)/lib/systemd/\" \
 	-DRANDOM_SEED=\"$(localstatedir)/lib/systemd/random-seed\" \
-	-DSYSTEMD_CRYPTSETUP_PATH=\"$(rootlibexecdir)/systemd-cryptsetup\" \
+	-DSYSTEMD_CRYPTSETUP_PATH=\"$(rootlibexecdir)/cryptsetup\" \
 	-DSYSTEM_GENERATOR_PATH=\"$(systemgeneratordir)\" \
 	-DUSER_GENERATOR_PATH=\"$(usergeneratordir)\" \
 	-DSYSTEM_SHUTDOWN_PATH=\"$(systemshutdowndir)\" \
--- systemd-221.orig/units/kmod-static-nodes.service.in
+++ systemd-221/units/kmod-static-nodes.service.in
@@ -15,4 +15,4 @@ ConditionPathExists=/lib/modules/%v/modu
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=@KMOD@ static-nodes --format=tmpfiles --output=/run/tmpfiles.d/kmod.conf
+ExecStart=/bin/kmod static-nodes --format=tmpfiles --output=/run/tmpfiles.d/kmod.conf
--- systemd-221.orig/units/systemd-fsck-root.service.in
+++ systemd-221/units/systemd-fsck-root.service.in
@@ -15,5 +15,5 @@ ConditionPathIsReadWrite=!/
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=@rootlibexecdir@/systemd-fsck
+ExecStart=@rootlibexecdir@/fscheck
 TimeoutSec=0
--- systemd-221.orig/units/systemd-fsck@.service.in
+++ systemd-221/units/systemd-fsck@.service.in
@@ -16,5 +16,5 @@ Before=shutdown.target
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=@rootlibexecdir@/systemd-fsck %f
+ExecStart=@rootlibexecdir@/fscheck %f
 TimeoutSec=0
--- systemd-221.orig/units/systemd-initctl.service.in
+++ systemd-221/units/systemd-initctl.service.in
@@ -11,5 +11,5 @@ Documentation=man:systemd-initctl.servic
 DefaultDependencies=no
 
 [Service]
-ExecStart=@rootlibexecdir@/systemd-initctl
+ExecStart=@rootlibexecdir@/initctl
 NotifyAccess=all
--- systemd-221.orig/units/systemd-journald.service.in
+++ systemd-221/units/systemd-journald.service.in
@@ -16,7 +16,7 @@ Before=sysinit.target
 [Service]
 Type=notify
 Sockets=systemd-journald.socket systemd-journald-dev-log.socket systemd-journald-audit.socket
-ExecStart=@rootlibexecdir@/systemd-journald
+ExecStart=@rootlibexecdir@/journald
 Restart=always
 RestartSec=0
 NotifyAccess=all
--- systemd-221.orig/units/systemd-logind.service.in
+++ systemd-221/units/systemd-logind.service.in
@@ -19,7 +19,7 @@ Wants=dbus.socket
 After=dbus.socket
 
 [Service]
-ExecStart=@rootlibexecdir@/systemd-logind
+ExecStart=@rootlibexecdir@/logind
 Restart=always
 RestartSec=0
 BusName=org.freedesktop.login1
--- systemd-221.orig/units/systemd-modules-load.service.in
+++ systemd-221/units/systemd-modules-load.service.in
@@ -23,5 +23,5 @@ ConditionKernelCommandLine=|rd.modules-l
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=@rootlibexecdir@/systemd-modules-load
+ExecStart=@rootlibexecdir@/modules-load
 TimeoutSec=90s
--- systemd-221.orig/units/systemd-remount-fs.service.in
+++ systemd-221/units/systemd-remount-fs.service.in
@@ -19,4 +19,4 @@ ConditionPathExists=/etc/fstab
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=@rootlibexecdir@/systemd-remount-fs
+ExecStart=@rootlibexecdir@/remount-fs
--- systemd-221.orig/units/systemd-sysctl.service.in
+++ systemd-221/units/systemd-sysctl.service.in
@@ -17,5 +17,5 @@ ConditionPathIsReadWrite=/proc/sys/
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=@rootlibexecdir@/systemd-sysctl
+ExecStart=@rootlibexecdir@/sysctl-bridge
 TimeoutSec=90s
--- systemd-221.orig/units/systemd-tmpfiles-clean.service.in
+++ systemd-221/units/systemd-tmpfiles-clean.service.in
@@ -15,5 +15,5 @@ Before=shutdown.target
 
 [Service]
 Type=oneshot
-ExecStart=@rootbindir@/systemd-tmpfiles --clean
+ExecStart=@rootbindir@/tmpfiles --clean
 IOSchedulingClass=idle
--- systemd-221.orig/units/systemd-tmpfiles-setup-dev.service.in
+++ systemd-221/units/systemd-tmpfiles-setup-dev.service.in
@@ -17,4 +17,4 @@ ConditionCapability=CAP_SYS_MODULE
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=@rootbindir@/systemd-tmpfiles --prefix=/dev --create --boot
+ExecStart=@rootbindir@/tmpfiles --prefix=/dev --create --boot
--- systemd-221.orig/units/systemd-tmpfiles-setup.service.in
+++ systemd-221/units/systemd-tmpfiles-setup.service.in
@@ -17,4 +17,4 @@ RefuseManualStop=yes
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=@rootbindir@/systemd-tmpfiles --create --remove --boot --exclude-prefix=/dev
+ExecStart=@rootbindir@/tmpfiles --create --remove --boot --exclude-prefix=/dev
--- systemd-221.orig/units/systemd-update-utmp-runlevel.service.in
+++ systemd-221/units/systemd-update-utmp-runlevel.service.in
@@ -18,4 +18,4 @@ Before=shutdown.target
 
 [Service]
 Type=oneshot
-ExecStart=@rootlibexecdir@/systemd-update-utmp runlevel
+ExecStart=@rootlibexecdir@/update-utmp runlevel
--- systemd-221.orig/units/systemd-update-utmp.service.in
+++ systemd-221/units/systemd-update-utmp.service.in
@@ -17,5 +17,5 @@ Before=sysinit.target shutdown.target
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=@rootlibexecdir@/systemd-update-utmp reboot
-ExecStop=@rootlibexecdir@/systemd-update-utmp shutdown
+ExecStart=@rootlibexecdir@/update-utmp reboot
+ExecStop=@rootlibexecdir@/update-utmp shutdown
--- systemd-221.orig/units/systemd-user-sessions.service.in
+++ systemd-221/units/systemd-user-sessions.service.in
@@ -13,5 +13,5 @@ After=remote-fs.target nss-user-lookup.t
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=@rootlibexecdir@/systemd-user-sessions start
-ExecStop=@rootlibexecdir@/systemd-user-sessions stop
+ExecStart=@rootlibexecdir@/user-sessions start
+ExecStop=@rootlibexecdir@/user-sessions stop
